src/shogun/features/RandomKitchenSinksDotFeatures.cpp

```diff
@@ -35,16 +36,12 @@ RandomKitchenSinksDotFeatures::RandomKitchenSinksDotFeatures(
 
 SGMatrix<float64_t> RandomKitchenSinksDotFeatures::generate_random_coefficients()
 {
-	SGVector<float64_t> vec = generate_random_parameter_vector();
-	SGMatrix<float64_t> random_params(vec.vlen, num_samples);
-	for (index_t dim=0; dim<random_params.num_rows; dim++)
-		random_params(dim, 0) = vec[dim];
+	SGMatrix<float64_t> random_params(num_samples + 1, num_samples);
 
-	for (index_t sample=1; sample<num_samples; sample++)
+	for (index_t sample=0; sample<num_samples; sample++)
 	{
-		vec = generate_random_parameter_vector();
-		for (index_t dim=0; dim<random_params.num_rows; dim++)
-			random_params(dim, sample) = vec[dim];
+		auto vec = generate_random_parameter_vector();
```

Comment:
technically we could even use `const auto&` here to avoid one instantiation :) @gf712 ?

Reply:
yes, we could, and maybe even should!

Reply:
we will :)

Reply:
does the set_column method then need to accept const references?

Reply:
yes